hardware_interface/src/resource_manager.cpp

```diff
@@ -91,6 +91,22 @@ std::string interfaces_to_string(
   return ss.str();
 };
 
+void get_hardware_related_interfaces(
+  const std::vector<std::string> & hw_command_itfs,
+  const std::vector<std::string> & start_stop_interfaces_list,
+  std::vector<std::string> & hw_interfaces)
+{
+  hw_interfaces.clear();
```

Comment:
I think we could already `.reserve()` for the upper boundary of the size of `hw_interfaces`
maybe best done as a follow-up